src/input/plugins/YoutubeInputPlugin.cxx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2003-2020 The Music Player Daemon Project
+ * http://www.musicpd.org
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "YoutubeInputPlugin.hxx"
+#include "CurlInputPlugin.hxx"
+#include "PluginUnavailable.hxx"
+#include "../InputPlugin.hxx"
+#include "../InputStream.hxx"
+#include "../TaggedInputStream.hxx"
+#include "Chrono.hxx"
+#include "Log.hxx"
+#include "tag/Builder.hxx"
+#include "util/ScopeExit.hxx"
+#include "util/ExecOpen.hxx"
+#include "util/Domain.hxx"
+
+#include <yajl/yajl_tree.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+static Domain youtube_domain("youtube");
+
+static const char *input_youtube_prefixes[] = {
+	"https",
+	nullptr
+};
+
+static void
+input_youtube_init(EventLoop &, const ConfigBlock &)
+{
+	if(WEXITSTATUS(system("youtube-dl --version > /dev/null")) != 0)
```

Comment:
`system()` should never be used; spawning child processes really is complicated, for example `system()` doesn't restore the signal mask and many other things.